### PR TITLE
Setup React SPA App

### PR DESCRIPTION
**Pull request by hochiw**
- Initialized basic react app with planned structure
- Setup linters and translation sorters that can be executed from Makefile
- Updated README

refs #1

### CODE REVIEW

**Review by chihimng — CHANGES_REQUESTED** [review]

File: react-app/README.md [hunk 1]
```diff
@@ -1,46 +1,57 @@
...
+### i18n
 
-Instead, it will copy all the configuration files and the transitive dependencies (webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.
+1. Add you translation key & string in the [translation files](./src/i18n/translations)
+1. Translation key is in the format of `screenName.section.view.content`
+1. Use `<LocalizedText>` or `translate` from `useIntl()`
+1. Sort the translation by `make sort-translations`
 
-You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
+### Theming
```
[the remark this answers is not in the document]
Comment (hochiw) [reply react-app/README.md:55]: let's add this when we actually have it

File: react-app/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      application/javascript
+      application/x-javascript
+      image/svg+xml;
+
+  location / {
+    root   /usr/share/nginx/html;
+    index  index.html index.htm;
+    try_files $uri $uri/ /index.html;
+  }
+}
```
Comment (chihimng) [inline react-app/nginx.conf:23]: will we need to serve `config.js`, `robots.txt` and a ping endpoint?
ref https://github.com/oursky/project-house/blob/939893fdc86f94cf7a05fab77e830740c264375a/web/nginx.conf#L63

File: react-app/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM node:16.15.0 as builder
+
+WORKDIR /usr/src/graphql-schema
+COPY ./graphql-schema ./
+
+WORKDIR /usr/src/app
```
Comment (chihimng) [inline react-app/Dockerfile:6]: move yarn step before graphql-schema step for caching?

probably gql will be changed more often then packages list

**Review by chihimng — CHANGES_REQUESTED** [review]
otherwise lgtm

File: react-app/nginx.conf (hunk 2 above)
Comment (hochiw) [reply react-app/nginx.conf:23]: I think I will handle this in the deployment pr

File: react-app/Dockerfile (hunk 3 above)
Comment (hochiw) [reply react-app/Dockerfile:3]: @chihimng  moved

**Review by chihimng — APPROVED** [review]
will re-review entire project setup after all pr merged

**Comment by hochiw** [comment]
pushed changes for the e0072dd3ea2d0d2a7fdc7bd47e994ea31eaee142 commit to fix prettier issues.

**Pull request merged**